metricbeat/module/rabbitmq/node/node.go

```diff
@@ -40,20 +50,61 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 		return nil, err
 	}
 	http.SetHeader("Accept", "application/json")
+	nodesURL, _ := url.Parse(base.HostData().SanitizedURI)
+
+	overview, err := helper.NewHTTP(base)
+	if err != nil {
+		return nil, err
+	}
+	overviewURL, _ := nodesURL.Parse(overviewRelativePath)
+	overview.SetURI(overviewURL.String())
+	overview.SetHeader("Accept", "application/json")
 
 	return &MetricSet{
-		base,
-		http,
+		BaseMetricSet: base,
+		HTTP:          http,
+		nodesURL:      nodesURL,
+		overview:      overview,
 	}, nil
 }
 
-func (m *MetricSet) Fetch() ([]common.MapStr, error) {
-	content, err := m.HTTP.FetchContent()
+type apiOverview struct {
```

Comment:
We normally keep all the mapping code in `data.json`. But happy to change this convention :-)